Modules/xASL_wrp_BIDS2Legacy.m

```diff
@@ -65,12 +65,17 @@
 [~, studyName] = fileparts(x.dir.DatasetRoot);
 fprintf('Converting from BIDS to Legacy: %s   \n', studyName);
 
-%% 2. Define SubjectSession
-for iSubjSess=1:numel(x.modules.bids2legacy.BIDS.subjects) 
-    % Iterate over x.modules.bids2legacy.BIDS.subjects (indices that include both subjects & sessions)
-    % so  1 subject  6 session/visits, will give numel(x.modules.bids2legacy.BIDS.subjects)=6
-    % and 6 subjects 1 session/visits, will give numel(x.modules.bids2legacy.BIDS.subjects)=6
-    xASL_TrackProgress(iSubjSess, nSubjects);
+%% 2. Define Subject
+subjectBIDS = ['sub-' x.SUBJECT];
+
+
+indicesCurrentSubject = find(cellfun( @(y) strcmp(y, subjectBIDS), {x.modules.bids2legacy.BIDS.subjects.name}));
```

Comment:
Just to understand. Previously, we converted all subjects here. But probably this function was called for each subject.

So previously - for all subjects we tried converting all subjects resulting this super-long runtimes.
Now - we run this function for all subject, but convert only one subject at a time.

Am I right? It is perfect that you fix this bug. I just want to make sure that this is the reason.

Reply:
Yes!